### ISSUE TITLE
Errors parsing inline metadata

### ISSUE THREAD

**Issue by natelandau** [natelandau/obsidian-metadata#28]
Every time I run `obsidian-metadata` from the terminal, I am seeing this error, not sure how to fix it. Any help would be appreciated,

```bash
Error in sys.excepthook:
Traceback (most recent call last):
  File "/Users/drew/.asdf/installs/python/3.11.2/lib/python3.11/site-packages/typer/main.py", line 72, in except_hook
    rich_tb = Traceback.from_exception(
              ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/drew/.asdf/installs/python/3.11.2/lib/python3.11/site-packages/rich/traceback.py", line 335, in from_exception
    rich_traceback = cls.extract(
                     ^^^^^^^^^^^^
  File "/Users/drew/.asdf/installs/python/3.11.2/lib/python3.11/site-packages/rich/traceback.py", line 449, in extract
    locals={
           ^
  File "/Users/drew/.asdf/installs/python/3.11.2/lib/python3.11/site-packages/rich/traceback.py", line 450, in <dictcomp>
    key: pretty.traverse(
         ^^^^^^^^^^^^^^^^
  File "/Users/drew/.asdf/installs/python/3.11.2/lib/python3.11/site-packages/rich/pretty.py", line 852, in traverse
    node = _traverse(_object, root=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/drew/.asdf/installs/python/3.11.2/lib/python3.11/site-packages/rich/pretty.py", line 647, in _traverse
    args = list(iter_rich_args(rich_repr_result))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/drew/.asdf/installs/python/3.11.2/lib/python3.11/site-packages/rich/pretty.py", line 614, in iter_rich_args
    for arg in rich_args:
  File "/Users/drew/.asdf/installs/python/3.11.2/lib/python3.11/site-packages/obsidian_metadata/models/vault.py", line 86, in __rich_repr__
    yield "num_notes_in_scope", len(self.notes_in_scope)
                                    ^^^^^^^^^^^^^^^^^^^
AttributeError: 'Vault' object has no attribute 'notes_in_scope'

Original exception was:
Traceback (most recent call last):
  File "/Users/drew/.asdf/installs/python/3.11.2/bin/obsidian-metadata", line 8, in <module>
    sys.exit(app())
             ^^^^^
  File "/Users/drew/.asdf/installs/python/3.11.2/lib/python3.11/site-packages/typer/main.py", line 328, in __call__
    raise e
  File "/Users/drew/.asdf/installs/python/3.11.2/lib/python3.11/site-packages/typer/main.py", line 311, in __call__
    return get_command(self)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/drew/.asdf/installs/python/3.11.2/lib/python3.11/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/drew/.asdf/installs/python/3.11.2/lib/python3.11/site-packages/typer/core.py", line 716, in main
    return _main(
           ^^^^^^
  File "/Users/drew/.asdf/installs/python/3.11.2/lib/python3.11/site-packages/typer/core.py", line 216, in _main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/Users/drew/.asdf/installs/python/3.11.2/lib/python3.11/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/drew/.asdf/installs/python/3.11.2/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/drew/.asdf/installs/python/3.11.2/lib/python3.11/site-packages/typer/main.py", line 683, in wrapper
    return callback(**use_params)  # type: ignore
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/drew/.asdf/installs/python/3.11.2/lib/python3.11/site-packages/obsidian_metadata/cli.py", line 157, in main
    application.application_main()
  File "/Users/drew/.asdf/installs/python/3.11.2/lib/python3.11/site-packages/obsidian_metadata/models/application.py", line 48, in application_main
    self._load_vault()
  File "/Users/drew/.asdf/installs/python/3.11.2/lib/python3.11/site-packages/obsidian_metadata/models/application.py", line 37, in _load_vault
    self.vault: Vault = Vault(config=self.config, dry_run=self.dry_run)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/drew/.asdf/installs/python/3.11.2/lib/python3.11/site-packages/obsidian_metadata/models/vault.py", line 70, in __init__
    self.all_notes: list[Note] = [
                                 ^
  File "/Users/drew/.asdf/installs/python/3.11.2/lib/python3.11/site-packages/obsidian_metadata/models/vault.py", line 71, in <listcomp>
    Note(note_path=p, dry_run=self.dry_run) for p in self.all_note_paths
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/drew/.asdf/installs/python/3.11.2/lib/python3.11/site-packages/obsidian_metadata/models/notes.py", line 64, in __init__
    self.inline_metadata: InlineMetadata = InlineMetadata(self.file_content)
                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/drew/.asdf/installs/python/3.11.2/lib/python3.11/site-packages/obsidian_metadata/models/metadata.py", line 380, in __init__
    self.dict: dict[str, list[str]] = self._grab_inline_metadata(file_content)
                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/drew/.asdf/installs/python/3.11.2/lib/python3.11/site-packages/obsidian_metadata/models/metadata.py", line 407, in _grab_inline_metadata
    for k, v in stripped_null_values:
        ^^^^
ValueError: not enough values to unpack (expected 2, got 1)
```

_Originally posted by @Andyspickle in https://github.com/natelandau/obsidian-metadata/issues/20#issuecomment-1487789815_

**Comment by natelandau** [comment]
@Andyspickle please upgrade the package to version `0.11.1` and try running it again. This update is unlikely to fix the problem but includes better reporting of errors when parsing inline metadata. When it fails, please review the note it flags as having issues for anything obvious on your side.  If you don't see an issue, post the output again.  

thanks for helping me test this package. It's very hard to test every eventuality as we all use Obsidian differently.  😄

**Comment by Andyspickle** [comment]
So I found one file that was blocking it, which had an inline field at the end of the file called `Related Projects::` with no value (only whitespace). However, I was able to get past the error by adding a new line.

I also noticed that any inline field with a space in the name was picking up incorrectly.
eg. `Projects` instead of `Related Projects`. 

I have also faced another issue with inline fields, if there is a duplicate inline, it will error. 
For example, I have one note that has `up::` multiple times in the document.
